client/src/main/java/io/pravega/client/netty/impl/FlowHandler.java

```diff
@@ -96,6 +97,33 @@ public void closeFlow(ClientConnection clientConnection) {
         flowIdReplyProcessorMap.remove(flow);
     }
 
+    /**
+     * Create a Batch size tracker on existing connection.
+     *
+     * @param id identifier.
+     */
+    public void createAppendBatchSizeTracker(final UUID id) {
+        Exceptions.checkNotClosed(closed.get(), this);
+        if (idBatchSizeTrackerMap.containsKey(id)) {
+            log.info("Reusing Batch size tracker for id {}.", id);
+        } else {
+            log.info("Creating Batch size tracker for id {}.", id);
+            idBatchSizeTrackerMap.put(id, new AppendBatchSizeTrackerImpl());
+        }
+    }
+
+
+    /**
+     * get a Batch size tracker on existing connection.
+     *
+     * @param id identifier.
+     * @return Batch size Tracker object.
```

Comment:
What if the id isn't registered?

Reply:
now .. i changed the prototype of this method. now , flow id needs to passed.
if its not existing then it returns null.